test-d/formio/components/currency.test-d.ts

```diff
@@ -0,0 +1,145 @@
+import {expectAssignable, expectNotAssignable} from 'tsd';
+
+import {CurrencyComponentSchema} from '../../../lib/';
+
+// minimal currency component schema
+expectAssignable<CurrencyComponentSchema>({
+  id: '123',
+  type: 'currency',
+  key: 'aCurrency',
+  label: 'A currency',
+  currency: 'EUR',
+} as const);
+
+// with additional, number-component specific properties
+expectAssignable<CurrencyComponentSchema>({
+  id: '123',
+  type: 'currency',
+  key: 'aCurrency',
+  label: 'A currency',
+  currency: 'EUR',
+  decimalLimit: 3,
+  allowNegative: true,
+} as const);
+
+// multiple false (implicit) and appropriate default value type
+expectAssignable<CurrencyComponentSchema>({
+  id: '123',
+  type: 'currency',
+  key: 'aCurrency',
+  label: 'A currency',
+  currency: 'EUR',
+  defaultValue: 3,
+} as const);
+
+// different component type
+expectNotAssignable<CurrencyComponentSchema>({
+  type: 'textfield',
+} as const);
+
+// wrong currency
+expectNotAssignable<CurrencyComponentSchema>({
+  id: '123',
+  type: 'currency',
+  key: 'aCurrency',
+  label: 'A currency',
+  currency: 'dummy',
+} as const);
+
+// using unsupported properties
+expectNotAssignable<CurrencyComponentSchema>({
+  id: '123',
+  type: 'currency',
+  key: 'aCurrency',
+  label: 'A currency',
+  currency: 'EUR',
+  hideLabel: true,
+} as const);
+
+// no multiple support -> no array defaultValue
```

Comment:
there is multiple support, but only if `multiple: true`

those tests appear to be missing